app/models/post.rb

```diff
@@ -1,17 +1,18 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User'
+  belongs_to :user, class_name: 'User'
   has_many :comments, foreign_key: 'posts_id'
   has_many :likes, foreign_key: 'posts_id'
 
-  def self.post_counter(user_id)
-    user = User.find(user_id)
-    if user.posts_count.nil?
-      user.posts_count = 1
-    else
-      user.posts_count += 1
-    end
+  after_save :update_post_counter
 
-    user.save
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_count, numericality: { only_integer: true }
+  validates :comments_count, comparison: { greater_than_or_equal_to: 0 }
+  validates :comments_count, numericality: { only_integer: true }
+  validates :comments_count, comparison: { greater_than_or_equal_to: 0 }
```

Comment:
- [ ] Here, `comments_count` got validation again, I think here we should validate `likes_count` here also.